tools/dotnet-linker/Steps/ConfigurationAwareStep.cs

```diff
@@ -21,5 +23,50 @@ protected void Report (List<Exception> exceptions)
 			// We can't just throw an exception or exit here, since there might be only warnings in the list of exceptions.
 			ErrorHelper.Show (exceptions);
 		}
+
+		protected virtual void TryProcess ()
+		{
+			base.Process ();
+		}
+
+		protected sealed override void Process ()
+		{
+			try {
+				TryProcess ();
+			} catch (Exception e) {
+				Report (e);
+				throw;
+			}
+		}
```

Comment:
might want to share code/pattern with `tools/linker/ExceptionalSubStep.cs` since it faced a similar issue